File: backend/clock.py
from PySide6.QtCore import QObject, Signal, QTimer
from datetime import datetime

class Clock(QObject):
    timeChanged = Signal(str)
    
    def __init__(self, settings_manager):
        super().__init__()
        self._settings_manager = settings_manager
        self.timer = QTimer()
        self.timer.timeout.connect(self.update_time)
        self.timer.start(1000)  # Update every second
        
    def update_time(self):
        if not self._settings_manager.showClock:
            self.timeChanged.emit("")
            return
            
        current_time = datetime.now()
        if self._settings_manager.clockFormat24Hour:
            time_str = current_time.strftime("%H:%M")
        else:
            hour_min = current_time.strftime("%I:%M")
            am_pm = current_time.strftime("%p").upper()  # Force uppercase
            time_str = f"{hour_min} {am_pm}"
            
        self.timeChanged.emit(time_str)

File: src/managers/clock.cpp
#include "clock.h"
#include "settingsmanager.h"

#include <QDateTime>
#include <QTime>

Clock::Clock(SettingsManager *settingsManager, QObject *parent)
    : QObject(parent)
    , m_settingsManager(settingsManager)
{
    connect(&m_timer, &QTimer::timeout, this, &Clock::update_time);
    m_timer.start(1000); // Update every second
}

void Clock::update_time()
{
    if (!m_settingsManager->showClock()) {
        emit timeChanged(QString());
        return;
    }

    QTime currentTime = QTime::currentTime();
    const bool showSeconds = m_settingsManager->clockShowSeconds();

    if (m_settingsManager->clockFormat24Hour()) {
        const QString fmt = showSeconds ? QStringLiteral("HH:mm:ss") : QStringLiteral("HH:mm");
        emit timeChanged(currentTime.toString(fmt));
    } else {
        int hour12 = currentTime.hour() % 12;
        if (hour12 == 0)
            hour12 = 12;
        QString hourMin = QStringLiteral("%1:%2")
                              .arg(hour12, 2, 10, QLatin1Char('0'))
                              .arg(currentTime.minute(), 2, 10, QLatin1Char('0'));
        if (showSeconds) {
            hourMin += QStringLiteral(":%1").arg(currentTime.second(), 2, 10, QLatin1Char('0'));
        }
        const QString amPm = (currentTime.hour() >= 12) ? QStringLiteral("PM") : QStringLiteral("AM");
        emit timeChanged(hourMin + QStringLiteral(" ") + amPm);
    }
}
